Serialization error creating org folder vault credentials through JobDSL

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  <!-- Paste your environment details below -->
  ```text
Jenkins: 2.361.2
OS: Linux - 4.12.14-122.116-default
---
PrioritySorter:4.1.0
ace-editor:1.1
active-directory:2.26
ansicolor:1.0.2
ant:481.v7b_09e538fcca
antisamy-markup-formatter:2.7
apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61
artifactory:3.17.1
authentication-tokens:1.4
azure-ad:267.v5b_dfb_514d9fd
azure-sdk:118.v43f74dd9ca_dc
basic-branch-build-strategies:1.3.2
bitbucket:223.vd12f2bca5430
blueocean:1.25.8
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.25.8
blueocean-commons:1.25.8
blueocean-config:1.25.8
blueocean-core-js:1.25.8
blueocean-dashboard:1.25.8
blueocean-display-url:2.4.1
blueocean-events:1.25.8
blueocean-git-pipeline:1.25.8
blueocean-github-pipeline:1.25.8
blueocean-i18n:1.25.8
blueocean-jwt:1.25.8
blueocean-personalization:1.25.8
blueocean-pipeline-api-impl:1.25.8
blueocean-pipeline-editor:1.25.8
blueocean-pipeline-scm-api:1.25.8
blueocean-rest:1.25.8
blueocean-rest-impl:1.25.8
blueocean-web:1.25.8
bootstrap5-api:5.2.1-3
bouncycastle-api:2.26
branch-api:2.1046.v0ca_37783ecc5
build-discarder:139.v05696a_7fe240
build-user-vars-plugin:1.9
build-with-parameters:1.6
buildtriggerbadge:251.vdf6ef853f3f5
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
checks-api:1.7.5
cloudbees-bitbucket-branch-source:791.vb_eea_a_476405b
cloudbees-folder:6.758.vfd75d09eea_a_1
command-launcher:1.2
commons-lang3-api:3.12.0-36.vd97de6465d5b_
commons-text-api:1.9-19.v8df45c678366
config-file-provider:3.11.1
configuration-as-code:1512.vb_79d418d5fc8
credentials:1189.vf61b_a_5e2f62e
credentials-binding:523.vd859a_4b_122e6
display-url-api:2.3.6
docker-commons:1.21
docker-java-api:3.2.13-37.vf3411c9828b9
docker-plugin:1.2.10
docker-workflow:521.v1a_a_dd2073b_2e
durable-task:500.v8927d9fd99d8
echarts-api:5.4.0-1
email-ext:2.91
embeddable-build-status:255.va_d2370ee8fde
envinject:2.881.v37c62073ff97
envinject-api:1.199.v3ce31253ed13
favorite:2.4.1
font-awesome-api:6.2.0-3
git:4.12.1
git-client:3.12.1
github:1.35.0
github-api:1.303-400.v35c2d8258028
github-branch-source:1695.v88de84e9f6b_9
gradle:1.40
h2-api:1.4.199
handlebars:3.0.8
handy-uri-templates-2-api:2.1.8-22.v77d5b_75e6953
hashicorp-vault-plugin:356.ved18810a_b_828
htmlpublisher:1.31
http_request:1.16-adp1
instance-identity:116.vf8f487400980
ionicons-api:28.va_f3a_84439e5f
ivy:2.2
jackson2-api:2.13.3-285.vc03c0256d517
jakarta-activation-api:2.0.1-2
jakarta-mail-api:2.0.1-2
javadoc:226.v71211feb_e7e9
javax-activation-api:1.2.0-5
javax-mail-api:1.6.2-5
jaxb:2.3.6-2
jdk-tool:55.v1b_32b_6ca_f9ca
jenkins-design-language:1.25.8
jjwt-api:0.11.5-77.v646c772fddb_0
job-dsl:1.81
jquery3-api:3.6.1-2
jsch:0.1.55.61.va_e9ee26616e7
junit:1150.v5c2848328b_60
kubernetes:3724.v0920c1e0ec69
kubernetes-client-api:5.12.1-187.v577c3e368fb_6
kubernetes-credentials:0.9.0
kubernetes-credentials-provider:1.199.v4a_1d1f5d074f
ldap:2.12
lockable-resources:2.18
m2release:0.16.3
mailer:438.v02c7f0a_12fa_4
matrix-auth:3.1.5
matrix-project:785.v06b_7f47b_c631
maven-plugin:3.19
mercurial:1251.va_b_121f184902
metrics:4.2.10-389.v93143621b_050
momentjs:1.1.1
nodejs:1.5.1
okhttp-api:4.9.3-108.v0feda04578cf
pipeline-build-step:2.18
pipeline-graph-analysis:195.v5812d95a_a_2f9
pipeline-groovy-lib:612.v84da_9c54906d
pipeline-input-step:451.vf1a_a_4f405289
pipeline-maven:1205.vceea_7b_972817
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2114.v2654ca_721309
pipeline-model-definition:2.2114.v2654ca_721309
pipeline-model-extensions:2.2114.v2654ca_721309
pipeline-npm:0.9.2
pipeline-rest-api:2.24
pipeline-stage-step:296.v5f6908f017a_5
pipeline-stage-tags-metadata:2.2114.v2654ca_721309
pipeline-stage-view:2.24
pipeline-utility-steps:2.13.0
plain-credentials:139.ved2b_9cf7587b
plugin-util-api:2.18.0
popper2-api:2.11.6-2
pubsub-light:1.17
resource-disposer:0.20
scm-api:621.vda_a_b_055e58f7
script-security:1183.v774b_0b_0a_a_451
show-build-parameters:1.0
sidebar-link:2.2.0
simple-theme-plugin:103.va_161d09c38c7
snakeyaml-api:1.32-86.ve3f030a_75631
splunk-devops:1.10.0
splunk-devops-extend:1.10.0
sse-gateway:1.26
ssh-agent:295.v9ca_a_1c7cc3a_a_
ssh-credentials:305.v8f4381501156
ssh-slaves:2.846.v1b_70190624f5
sshd:3.236.ved5e1b_cb_50b_2
structs:324.va_f5d6774f3a_d
swarm:3.27
tekton-client:1.0.3
token-macro:308.v4f2b_ed62b_b_16
trilead-api:2.72.v2a_3236754f73
uno-choice:2.6.4
variant:59.vf075fe829ccb
windows-slaves:1.8.1
workflow-aggregator:590.v6a_d052e5a_a_b_5
workflow-api:1192.v2d0deb_19d212
workflow-basic-steps:994.vd57e3ca_46d24
workflow-cps:2802.v5ea_628154b_c2
workflow-durable-task-step:1199.v02b_9244f8064
workflow-job:1239.v71b_b_a_124a_725
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:838.va_3a_087b_4055b
ws-cleanup:0.43

  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

OS: Linux - 4.12.14-122.116-default
Using official docker controller and agent images.

### Reproduction steps

1. Create Job-DSL script to create organization folder with vault credentials
    <details>
      <summary>Example Job DSL Script</summary>

      ```groovy
      organizationFolder('FOO') {
          description('FOO Bitbucket Project')
          organizations {
              bitbucket {
                  credentialsId('bitbucket-orgfolder')
                  repoOwner('FOO')
                  serverUrl('https://example.com')
                  traits {
                      sourceRegexFilter {
                          regex('.*')
                      }
                      headRegexFilter {
                          regex('.*')
                      }
                      bitbucketBranchDiscovery {
                          strategyId(3)
                      }
                      bitbucketWebhookRegistration {
                          mode('ITEM')
                      }
                      bitbucketPullRequestDiscovery {
                          strategyId(1)
                      }
                  }
              }
          }
      
          properties {
              folderCredentialsProperty {
                  domainCredentials {
                      domainCredentials {
                          domain {
                              name('foo')
                              description('foo org folder')
                          }
                          credentials {
                              vaultUsernamePasswordCredentialImpl {
                                  scope('GLOBAL')
                                  id('example-user')
                                  description('example user account')
                                  path('kv/example-user')
                                  usernameKey('username')
                              }
                          }
                      }
                  }
              }
          }
      }
      ```
    </details>
2. Run job dsl script (first run actually works)
3. Run job dsl a second time to generate serialization error

### Expected Results

Job-DSL script executed successfully and created an organization folder with folder level vault credentials.

### Actual Results

Serialization error while executing the jod-dsl script.

```
java.lang.RuntimeException: Failed to serialize com.datapipe.jenkins.vault.credentials.common.AbstractVaultBaseStandardCredentials#context for class com.datapipe.jenkins.vault.credentials.common.VaultUsernamePasswordCredentialImpl
...
Caused by: com.thoughtworks.xstream.converters.ConversionException: Failed calling method
---- Debugging information ----
message             : Failed calling method
cause-exception     : java.lang.IllegalStateException
cause-message       : no parent set on jenkins.branch.OrganizationFolder[null]
method              : jenkins.branch.OrganizationFolder.writeReplace()
-------------------------------
	at com.thoughtworks.xstream.core.util.SerializationMembers.callWriteReplace(SerializationMembers.java:110)
	at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:156)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
	at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:285)
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:272)
	... 141 more
Caused by: java.lang.IllegalStateException: no parent set on jenkins.branch.OrganizationFolder[null]
	at hudson.model.AbstractItem.getParent(AbstractItem.java:211)
	at hudson.model.AbstractItem.getFullName(AbstractItem.java:482)
	at hudson.model.AbstractItem$Replacer.<init>(AbstractItem.java:636)
	at hudson.model.AbstractItem.lambda$writeReplace$0(AbstractItem.java:629)
	at hudson.XmlFile.replaceIfNotAtTopLevel(XmlFile.java:243)
	at hudson.model.AbstractItem.writeReplace(AbstractItem.java:629)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.thoughtworks.xstream.core.util.SerializationMembers.callWriteReplace(SerializationMembers.java:101)
```

<details>
  <summary>Full Stack Trace</summary>
  
```
java.lang.RuntimeException: java.io.IOException: java.lang.RuntimeException: Failed to serialize com.cloudbees.hudson.plugins.folder.AbstractFolder#properties for class jenkins.branch.OrganizationFolder
	at hudson.util.PersistedList._onModified(PersistedList.java:223)
	at hudson.util.PersistedList.add(PersistedList.java:82)
	at com.cloudbees.hudson.plugins.folder.AbstractFolder.addProperty(AbstractFolder.java:616)
	at jenkins.branch.OrganizationFolder.onLoad(OrganizationFolder.java:242)
	at hudson.model.AbstractItem$1.call(AbstractItem.java:906)
	at hudson.model.AbstractItem$1.call(AbstractItem.java:904)
	at hudson.model.Items.whileUpdatingByXml(Items.java:132)
	at hudson.model.AbstractItem.updateByXml(AbstractItem.java:904)
	at javaposse.jobdsl.plugin.JenkinsJobManagement.updateExistingItem(JenkinsJobManagement.java:482)
	at javaposse.jobdsl.plugin.JenkinsJobManagement.createOrUpdateConfig(JenkinsJobManagement.java:169)
	at javaposse.jobdsl.dsl.JobManagement$createOrUpdateConfig$2.call(Unknown Source)
	at javaposse.jobdsl.plugin.InterruptibleJobManagement.createOrUpdateConfig(InterruptibleJobManagement.groovy:35)
	at javaposse.jobdsl.dsl.JobManagement$createOrUpdateConfig$2.call(Unknown Source)
	at javaposse.jobdsl.dsl.AbstractDslScriptLoader$_extractGeneratedJobs_closure4.doCall(AbstractDslScriptLoader.groovy:204)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
	at groovy.lang.Closure.call(Closure.java:420)
	at groovy.lang.Closure.call(Closure.java:436)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2125)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2110)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2151)
	at org.codehaus.groovy.runtime.dgm$163.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:274)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
	at javaposse.jobdsl.dsl.AbstractDslScriptLoader.extractGeneratedJobs(AbstractDslScriptLoader.groovy:197)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:177)
	at javaposse.jobdsl.dsl.AbstractDslScriptLoader.extractGeneratedItems(AbstractDslScriptLoader.groovy:184)
	at javaposse.jobdsl.plugin.JenkinsDslScriptLoader.extractGeneratedItems(JenkinsDslScriptLoader.java:22)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:352)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:68)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at javaposse.jobdsl.dsl.AbstractDslScriptLoader$_runScripts_closure1.doCall(AbstractDslScriptLoader.groovy:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
	at groovy.lang.Closure.call(Closure.java:420)
	at groovy.lang.Closure.call(Closure.java:436)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2125)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2110)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2163)
	at org.codehaus.groovy.runtime.dgm$165.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:274)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
	at javaposse.jobdsl.dsl.AbstractDslScriptLoader.runScripts(AbstractDslScriptLoader.groovy:46)
	at javaposse.jobdsl.plugin.ExecuteDslScripts.perform(ExecuteDslScripts.java:351)
	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:101)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.IOException: java.lang.RuntimeException: Failed to serialize com.cloudbees.hudson.plugins.folder.AbstractFolder#properties for class jenkins.branch.OrganizationFolder
	at hudson.XmlFile.write(XmlFile.java:220)
	at hudson.model.AbstractItem.save(AbstractItem.java:620)
	at com.cloudbees.hudson.plugins.folder.AbstractFolder.save(AbstractFolder.java:1313)
	at hudson.util.PersistedList.onModified(PersistedList.java:190)
	at hudson.util.PersistedList._onModified(PersistedList.java:221)
	... 81 more
Caused by: java.lang.RuntimeException: Failed to serialize com.cloudbees.hudson.plugins.folder.AbstractFolder#properties for class jenkins.branch.OrganizationFolder
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:276)
	at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:243)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:174)
	at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:228)
	at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:165)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
	at com.thoughtworks.xstream.core.TreeMarshaller.start(TreeMarshaller.java:83)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.marshal(AbstractTreeMarshallingStrategy.java:37)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:1266)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:1255)
	at com.thoughtworks.xstream.XStream.toXML(XStream.java:1228)
	at hudson.XmlFile.write(XmlFile.java:213)
	... 85 more
Caused by: java.lang.RuntimeException: Failed to serialize com.cloudbees.hudson.plugins.folder.properties.FolderCredentialsProvider$FolderCredentialsProperty#domainCredentialsMap for class com.cloudbees.hudson.plugins.folder.properties.FolderCredentialsProvider$FolderCredentialsProperty
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:276)
	at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:243)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:174)
	at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:228)
	at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:165)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeBareItem(AbstractCollectionConverter.java:94)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:66)
	at hudson.util.DescribableList$ConverterImpl.marshal(DescribableList.java:274)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
	at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:285)
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:272)
	... 98 more
Caused by: java.lang.RuntimeException: Failed to serialize com.datapipe.jenkins.vault.credentials.common.AbstractVaultBaseStandardCredentials#context for class com.datapipe.jenkins.vault.credentials.common.VaultUsernamePasswordCredentialImpl
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:276)
	at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:243)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:174)
	at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:228)
	at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:165)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeBareItem(AbstractCollectionConverter.java:94)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:66)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeCompleteItem(AbstractCollectionConverter.java:81)
	at com.thoughtworks.xstream.converters.collections.CollectionConverter.marshal(CollectionConverter.java:75)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeBareItem(AbstractCollectionConverter.java:94)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:66)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeCompleteItem(AbstractCollectionConverter.java:81)
	at com.thoughtworks.xstream.converters.collections.MapConverter.marshal(MapConverter.java:80)
	at hudson.util.CopyOnWriteMap$Hash$ConverterImpl.marshal(CopyOnWriteMap.java:209)
	at hudson.util.XStream2$AssociatedConverterImpl.marshal(XStream2.java:495)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
	at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:285)
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:272)
	... 114 more
Caused by: com.thoughtworks.xstream.converters.ConversionException: Failed calling method
---- Debugging information ----
message             : Failed calling method
cause-exception     : java.lang.IllegalStateException
cause-message       : no parent set on jenkins.branch.OrganizationFolder[null]
method              : jenkins.branch.OrganizationFolder.writeReplace()
-------------------------------
	at com.thoughtworks.xstream.core.util.SerializationMembers.callWriteReplace(SerializationMembers.java:110)
	at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:156)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
	at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:285)
	at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:272)
	... 141 more
Caused by: java.lang.IllegalStateException: no parent set on jenkins.branch.OrganizationFolder[null]
	at hudson.model.AbstractItem.getParent(AbstractItem.java:211)
	at hudson.model.AbstractItem.getFullName(AbstractItem.java:482)
	at hudson.model.AbstractItem$Replacer.<init>(AbstractItem.java:636)
	at hudson.model.AbstractItem.lambda$writeReplace$0(AbstractItem.java:629)
	at hudson.XmlFile.replaceIfNotAtTopLevel(XmlFile.java:243)
	at hudson.model.AbstractItem.writeReplace(AbstractItem.java:629)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.thoughtworks.xstream.core.util.SerializationMembers.callWriteReplace(SerializationMembers.java:101)
	... 147 more
```
</details>

### Anything else?

This seems to be associated with the addition of `ItemGroup` to `AbstractVaultBaseStandardCredentials` #195

--- a/src/main/java/com/datapipe/jenkins/vault/credentials/common/AbstractVaultBaseStandardCredentials.java
+++ b/src/main/java/com/datapipe/jenkins/vault/credentials/common/AbstractVaultBaseStandardCredentials.java
@@ -23,7 +23,7 @@ public abstract class AbstractVaultBaseStandardCredentials extends BaseStandardC
     private String prefixPath;
     private String namespace;
     private Integer engineVersion;
-    private ItemGroup context;
+    private transient ItemGroup context;
 
     AbstractVaultBaseStandardCredentials(CredentialsScope scope, String id, String description) {
         super(scope, id, description);

--- a/src/test/java/com/datapipe/jenkins/vault/credentials/common/AbstractVaultBaseStandardCredentialsTest.java
+++ b/src/test/java/com/datapipe/jenkins/vault/credentials/common/AbstractVaultBaseStandardCredentialsTest.java
@@ -1,0 +1,52 @@
+package com.datapipe.jenkins.vault.credentials.common;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import hudson.XmlFile;
+import hudson.model.ItemGroup;
+import java.io.File;
+import java.io.FileInputStream;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathFactory;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.Issue;
+import org.w3c.dom.Document;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+public class AbstractVaultBaseStandardCredentialsTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    /**
+     * Verify {@link AbstractVaultBaseStandardCredentials} does not attempt to serialize ItemGroup
+     * context.
+     */
+    @Test
+    @Issue("https://github.com/jenkinsci/hashicorp-vault-plugin/issues/264")
+    public void itemGroupContextNotSerialized() throws Exception {
+        ItemGroup ig = mock(ItemGroup.class);
+        VaultUsernamePasswordCredentialImpl cred = new VaultUsernamePasswordCredentialImpl(
+            CredentialsScope.GLOBAL, "foo", "foo credential");
+        cred.setContext(ig);
+
+        // serialize object using xstream
+        File out = folder.newFile();
+        new XmlFile(out).write(cred);
+
+        // verify document does not include include context field
+        try (FileInputStream fis = new FileInputStream(out)) {
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            Document doc = db.parse(fis);
+            XPath xpath = XPathFactory.newInstance().newXPath();
+            String expression = "boolean(/com.datapipe.jenkins.vault.credentials.common.VaultUsernamePasswordCredentialImpl/context/node())";
+            assertEquals("false", xpath.compile(expression).evaluate(doc));
+        }
+    }
+}

